backend/feed_service.py

```
@@ -27,12 +30,548 @@
+    logger.info(
+        "OPML import: Invalidated caches for tabs: %s.",
+        affected_tab_ids_set,
+    )
+
+
+def import_opml(opml_file_stream, requested_tab_id_str):
+    """Imports feeds from an OPML file, sending progress via SSE."""
+    # Read the entire stream to a variable, so we can parse it multiple times
+    try:
+        opml_content = opml_file_stream.read()
+    except Exception as e:
```

Ruff (0.14.14)

Do not catch blind exception: `Exception`

(BLE001)